service-06/PUS-6-1.asn1

```diff
@@ -0,0 +1,55 @@
+--!
+--! Copyright (C) 2017 N7 Mobile sp. z o. o.
+--! Contact: http://n7mobile.pl/Space
+--!
+--! This file is part of ASN.1/ACN PUS-C Components Library.
+--!
+--! Library was developed under a programme and funded by
+--! European Space Agency.
+--!
+--! This Library is free software: you can redistribute it and/or modify
+--! it under the terms of the GNU General Public License as published by
+--! the Free Software Foundation, either version 3 of the License, or
+--! (at your option) any later version.
+--!
+--! This Library is distributed in the hope that it will be useful,
+--! but WITHOUT ANY WARRANTY; without even the implied warranty of
+--! MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+--! GNU General Public License for more details.
+--!
+--! You should have received a copy of the GNU General Public License
+--! along with this program.  If not, see <http://www.gnu.org/licenses/>.
+--!
+
+PUS-6-1 DEFINITIONS AUTOMATIC TAGS ::= BEGIN
+EXPORTS ALL;
+IMPORTS
+    Base, BaseMultipleMemid FROM StructuredMemory
+    Memory-ID FROM Memory
+    MemoryData FROM Data
+    PUSC-UINT32 FROM BasicTypes;
+
+maxTC-6-1-LoadObjectInstructionCount INTEGER ::= 2147483647
+
+TC-6-1-LoadObjectMemoryData ::= TC-6-1-LoadObjectMemoryDataGeneric {Base}
+
+TC-6-1-LoadObjectMemoryDataMultipleMemid ::= TC-6-1-LoadObjectMemoryDataGeneric {BaseMultipleMemid}
+
+LoadObjectInstruction ::= LoadObjectInstructionGeneric {PUSC-UINT32, MemoryData}
+
+TC-6-1-LoadObjectMemoryDataGeneric {Base-Type} ::= SEQUENCE
+{
+--! memory-ID Memory-ID-Type, moved to acn
+    base-ID Base-Type,
+--! n UnsignedInteger-Type, moved to acn
+    loadObjectInstructions SEQUENCE (SIZE(0 .. maxTC-6-1-LoadObjectInstructionCount)) OF LoadObjectInstruction
```

Comment:
can it be `0`?

Reply:
and shouldn't LoadObjectInsctuction be a generic parameter?

Reply:
Yeah, should be > 1. 

LoadObjectInsctuction is constant part of this message. Only it's fields should be parameterized, not it's structure

Reply:
but that makes it possible parameter of current structure :) for example imagine system usng two different pus6 services for two different modules :)